docs/source/migration.md

```diff
@@ -0,0 +1,716 @@
+Code Migration from Intel Neural Compressor 1.X to Intel Neural Compressor 2.X
+===============
+
+Intel Neural Compressor is a powerful open-source Python library that runs on Intel CPUs and GPUs, which delivers unified interfaces across multiple deep-learning frameworks for popular network compression technologies such as quantization, pruning, and knowledge distillation. We conduct several changes to our old APIs based on Intel Neural Compressor 1.X to make it more user-friendly and convenient for using. Just some simple steps could upgrade your code from Intel Neural Compressor 1.X to Intel Neural Compressor 2.X.
+
+> It should be noted that we will stop to support Intel Neural Compressor 1.X in the future.
+
+1. [Quantization](#model-quantization)
+2. [Pruning](#pruning)
+3. [Distillation](#distillation)
+4. [Mix-Precision](#mix-precision)
+5. [Orchestration](#orchestration)
+6. [Benchmark](#benchmark)
+7. [Examples](#examples)
+
```

Comment:
For advanced user, how to find the map rule for the parameters in yaml which are not listed in this guide?
Is there any common method to fix it?
for example:
  neural_compressor.config.PostTrainingQuantConfig  <-> quantization in yaml file.

Reply:
The parameters that are listed in this guide do not influence the actual tuning performance. I followed the previous yaml template to map from 1.x to 2.x. In the configuration of 2.x, I have introduced the mapping relation one by one. It is very easy for the user to find the desired name of the parameters.